tests/periph_gpio/main.c

```diff
@@ -203,6 +207,97 @@ static int toggle(int argc, char **argv)
     return 0;
 }
 
+static void bench_print_time(uint32_t time, unsigned long runs, const char *name)
+{
+    uint32_t full = (time / runs);
+    uint32_t div  = (time - (full * runs)) / (runs / 1000);
+
+    printf("%11s: %9luus  ---  %2u.%03uus per call\n", name,
+           (unsigned long) time, (unsigned)full, (unsigned)div);
+}
+
+static int bench(int argc, char **argv)
+{
+    if (argc < 3) {
+        printf("usage: %s <port> <pin>\n", argv[0]);
+        return 1;
+    }
+
+    uint32_t time;
+    gpio_t pin = GPIO_PIN(atoi(argv[1]), atoi(argv[2]));
+    unsigned long runs = BENCH_RUNS_DEFAULT;
+    if (argc > 3) {
+        runs = (unsigned long)atoi(argv[3]);
```

Comment:
The return value needs to be checked (for >0). Otherwise, later on, this can lead to div by zero. (atio() returns 0 for non-number strings...)

Reply:
And maybe use atol() or atoll() so 16bitter can do more than 2^15?